enclave/core/sgx/backtrace.c

```diff
@@ -83,17 +73,20 @@ int oe_backtrace(void** buffer, int size)
     }
 
     return n;
-#else
-    return 0;
-#endif
 }
 
-char** oe_backtrace_symbols(void* const* buffer, int size)
+int oe_backtrace(void** buffer, int size)
+{
+    return oe_backtrace_impl(__builtin_frame_address(0), buffer, size);
```

Comment:
Should this function be marked to always be inlined?

Reply:
Yes, this function ought to have the behavior of being inlined so that oe_backtrace itself does not show up in the stack frames. However we don't want to put the implementation in a header file and want to keep it internal.

Instead we pass the oe_backtrace function's frame to oe_backtrace_impl which has the same effect as if we are jumping to oe_backtrace_impl. The logic within oe_backtrace_impl starts the walk at the callee of the passed frame and thus oe_backtrace is excluded from the trace.

Any suggestions for better organizing these two functions' responsibilities welcome.